test/lambdaisland/uri/normalize_test.cljc

```diff
@@ -21,6 +21,16 @@
     (uri/map->URI {:query "text=You are welcome 🙂"}) "?text=You%20are%20welcome%20%F0%9F%99%82"
     ))
 
+(deftest char-seq-test
+  (let [long-string (->> "s"
+                         ;; Long enough to trigger StackOverflow in non-tail recursive cases.
+                         (repeat 5000)
```

Comment:
I believe typically it's around 4-5K stack frames where things start to get potentially dangerous for overflow on the standard default JVM settings.